### PR TITLE
[node, cf-core, cf-types, client] Needed some fixes for the multisig issues, added the propose protocol

### PR DESCRIPTION
**Pull request by rhlsthrm**
## Checklist:

- [ ] I am making this PR against staging, not master
- [ ] My code follows the code style of this project.
- [ ] I have described any backwards-incompatibility implications above.
- [ ] I have highlighted which parts of the code should be reviewed most carefully.
- [ ] If my change requires a change to the documentation, I have updated it accordingly.

### CODE REVIEW

File: modules/cf-core/src/store.ts [hunk 1]
```diff
@@ -215,7 +216,8 @@ export class Store {
     myIdentifier: string,
     theirIdentifier: string,
     proxyFactoryAddress: string,
-    minimumViableMultisigAddress: string
+    minimumViableMultisigAddress: string,
+    acceptGeneratedMultisig: boolean // = false,
```
Comment (rhlsthrm) [inline modules/cf-core/src/store.ts]: Why not default to false?

File: modules/cf-core/src/protocol/install-virtual-app.ts [hunk 2]
```diff
@@ -320,7 +322,8 @@ export const INSTALL_VIRTUAL_APP_PROTOCOL: ProtocolExecutionFlow = {
       stateChannelsMap,
       (virtualAppInstanceIdentityHash as unknown) as string,
       (virtualAppInstanceDefaultOutcome as unknown) as string,
-      network
+      network,
+      store,
```
Comment (rhlsthrm) [inline modules/cf-core/src/protocol/install-virtual-app.ts]: Do we need this anymore?

File: modules/cf-core/src/store.ts (hunk 1 above)
Comment (LayneHaber) [reply modules/cf-core/src/store.ts]: that was temporary, added so the compiler would  point me to all uses. I wanted to go through them one by one in this case, but it now does default to false

File: modules/cf-core/src/protocol/install-virtual-app.ts (hunk 2 above)
Comment (LayneHaber) [reply modules/cf-core/src/protocol/install-virtual-app.ts]: nope, removed and using static fn

**Comment by LayneHaber** [comment]
alrighty so to squash the multisig address/proxy factory issues and prevent any issues coming up with bytecode things ive done the following:
- removed any calls to getCreate2MultisigAddress accept for within the store class
- added a static method to the store class so we can access this functionality without including the store in the protocol context
- add a boolean flag to the getMultisigAddressWithCounterparty to the store, if true it will accept the address returned from the getCreate2... fn, and if false, it will only take the values from the store
- set the flag to default false, EXCEPT in the following cases:
1. create
2. propose install (no clear indicator of whether it is a virtual or regular proposal, and since most initiator/responder multisigs will not exist we cant enforce it come from the store)
3. un- and install virtual for multisigs between all three participants, or initiator/responder (not intermediary) channels

**Pull request merged**